client/src/shared/ui/form/ToggleSwitch.js

```diff
@@ -31,7 +31,7 @@ export default function ToggleSwitch(props) {
             { label }
           </label>
           <div className="field-wrapper">
-            <label className="toggle-switch__switcher">
+            <label className="toggle-switch__switcher" onChange={ form.submitForm }>
```

Comment:
When I add an `onChange` handler to the toggle component, don't I already have the control to submit the form on the implementation side? Is there a reason we make this explicit here?

Reply:
If I just pass the handler to the toggle like so👇, the handler is not applied. Therefore, I need to add the `onChange` explicitly.
```javascript
<fieldset className="fieldset">
   <Field
      onChange={ formikProps.submitForm }
      component={ ToggleSwitch }
      (...)
    />
</fieldset>
```
What I could do is add it to the wrapper of the component 👇 but I thought focusing it on the actual component would be better.
```javascript
<fieldset className="fieldset"   onChange={ formikProps.submitForm }>
   <Field
      component={ ToggleSwitch }
      (...)
    />
</fieldset>
```

Reply:
I remember React had some difficulties with `onChange` handlers and checkboxes: https://stackoverflow.com/questions/26615779/react-checkbox-not-sending-onchange

The solution would be to use `defaultChecked` instead of `checked`. We use that in other places as well: https://github.com/camunda/camunda-modeler/blob/develop/client/src/plugins/privacy-preferences/PrivacyPreferencesView.js#L73

Can you try out whether that works? Another alternative could be to use `onClick` or `onInput`.

Maybe we don't need this change at all 🙂

Reply:
I replaced `checked` with `defaultChecked` and tested it by passing the `onChange` as props. This way it works without adding it explicitly inside the component.

Reply:
👍